pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/IsNotNullTransformFunction.java

```diff
@@ -0,0 +1,101 @@
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.core.operator.transform.function;
+
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import org.apache.pinot.core.operator.blocks.ProjectionBlock;
+import org.apache.pinot.core.operator.transform.TransformResultMetadata;
+import org.apache.pinot.segment.spi.datasource.DataSource;
+import org.apache.pinot.segment.spi.index.reader.NullValueVectorReader;
+import org.roaringbitmap.PeekableIntIterator;
+
+
+public class IsNotNullTransformFunction extends BaseTransformFunction {
+  public static final String FUNCTION_NAME = "IS_NOT_NULL";
+
+  private TransformFunction _leftTransformFunction;
+  private int[] _results;
+  private Map<String, DataSource> _dataSourceMap = new HashMap<>();
+  private PeekableIntIterator _nullValueVectorIterator;
+  private NullValueVectorReader _nullValueVectorReader;
+
+  @Override
+  public String getName() {
+    return FUNCTION_NAME;
+  }
+
+  @Override
+  public void init(List<TransformFunction> arguments, Map<String, DataSource> dataSourceMap) {
+    Preconditions.checkArgument(arguments.size() == 1,
+        "Exact 1 argument is required for IS_NOT_NULL operator function");
+    _leftTransformFunction = arguments.get(0);
+    if (!(_leftTransformFunction instanceof IdentifierTransformFunction)) {
+      throw new IllegalArgumentException(
+          "Only column names are supported in IS_NOT_NULL. Support for functions is planned for future release");
+    }
+    _dataSourceMap = dataSourceMap;
+    String columnName = ((IdentifierTransformFunction) _leftTransformFunction).getColumnName();
+    _nullValueVectorReader = _dataSourceMap.get(columnName).getNullValueVector();
+    if (_nullValueVectorReader != null) {
+      _nullValueVectorIterator = _nullValueVectorReader.getNullBitmap().getIntIterator();
+    } else {
+      _nullValueVectorIterator = null;
+    }
+  }
+
+  @Override
+  public TransformResultMetadata getResultMetadata() {
+    return BOOLEAN_SV_NO_DICTIONARY_METADATA;
+  }
+
+  @Override
+  public int[] transformToIntValuesSV(ProjectionBlock projectionBlock) {
+    int length = projectionBlock.getNumDocs();
+    if (_results == null || _results.length < length) {
+      _results = new int[length];
+    }
+
+    int[] docIds = projectionBlock.getDocIds();
+
+    if (!_nullValueVectorIterator.hasNext() || (length > 0 && _nullValueVectorIterator.peekNext() > docIds[0])) {
+      _nullValueVectorIterator = _nullValueVectorReader.getNullBitmap().getIntIterator();
+    }
```

Comment:
This part is redundant. The docIds passed in should have no overlap with the previous array.

Reply:
The unit tests were failing without it because I pass same projection block here
```java
  int[] intValues = transformFunction.transformToIntValuesSV(_projectionBlock);
  long[] longValues = transformFunction.transformToLongValuesSV(_projectionBlock);
  float[] floatValues = transformFunction.transformToFloatValuesSV(_projectionBlock);
  double[] doubleValues = transformFunction.transformToDoubleValuesSV(_projectionBlock);
  String[] stringValues = transformFunction.transformToStringValuesSV(_projectionBlock);
```     

Should I recreate transform function for each of these? This scenario can never occur in production code?

Reply:
Yes, each `ProjectionBlock` should be processed only per transform function. We should simplify the handling to avoid overhead, and fix the test instead

Reply:
done